### PR TITLE
Add coreference model

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Creating a pull request adding the corefence spacy component.

@HLasse this what I have added:
- [x] workflow for pypi install 
- [x] Add download functionality
- [x] Add tests
- [x] Add documentation
- [x] Add spacy component
- [x] Update requirements, including version numbers
- [x] Polish

Minor: 
- [x] format to black

Fix download link:
Stephan (github ID?) can you fix the download link such that it is on the form:
```
da_coref_twitter_v1
```
instead of the nested:
```
conlldata_danish_twitter_lr1_lr1/ser_folder
```
after that I can 
- [ ] remove current hotfix

### CODE REVIEW

File: src/conspiracies/coref/CoreferenceModel.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self, model_path: Union[Path, str, None] = None, device: int = -1, **kwargs
+    ) -> None:
+
+        if model_path is None:
+            model_path = download_model("da_coref_twitter_v1")
+
+        # TODO remove this hotfix
+        import os
+
+        model_path = os.path.join(model_path, "ser_folder")
```
Comment (KennethEnevoldsen) [inline src/conspiracies/coref/CoreferenceModel.py:47]: Should be removed before merging

File: src/conspiracies/download.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+from tqdm import tqdm
+from wasabi import msg
+
+
+MODEL_URLS = {
+    "da_coref_twitter_v1": {
+        "url": "https://sciencedata.dk/shared/3d0b045fcae482ff997b7ff6243361d1?download",
+        "folder_name": "conlldata_danish_twitter_lr1_lr1",
+    }
```
Comment (KennethEnevoldsen) [inline src/conspiracies/download.py:19]:
```suggestion
    "da_coref_twitter_v1": {
        "url": "https://sciencedata.dk/shared/f20703362e0a771f004f8ff2026afb0f?download",
        "folder_name": "da_coref_twitter_v1",
    }
```

File: src/conspiracies/coref/CoreferenceModel.py (hunk 1 above)
Comment (KennethEnevoldsen) [inline src/conspiracies/coref/CoreferenceModel.py:47]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -13,14 +13,51 @@ Discovering and examining conspiracies using NLP.
 
 
 ## 🔧 Installation
+Installation is simply using pip simply run:
```
Comment (HLasse) [inline README.md:16]:
```suggestion
Installation using pip:
```

File: src/conspiracies/coref/CoreferenceModel.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# required for reading an archive using the "coref"
+from allennlp_models.coref.dataset_readers.conll import ConllCorefReader  # noqa
+
+from conspiracies.download import download_model
+
+
+@Predictor.register("coreference_resolution_v1")
+class CoreferenceModel(Predictor):
+    """
+    A coreference predictor model using the AllenNLP API, but for handing SpaCy docs
```
Comment (HLasse) [inline src/conspiracies/coref/CoreferenceModel.py:25]:
```suggestion
    A coreference predictor model using the AllenNLP API, but for handling SpaCy docs
```

**Comment by KennethEnevoldsen** [comment]
Hmm there seems to be a problem with Windows path and in the allenNLP package as they [don't support windows](https://docs.allennlp.org/main/) I think it is fair to simply remove it from the tests.

**Pull request merged**